src/Features/VisualBasic/Portable/Completion/CompletionProviders/EnumCompletionProvider.vb

```diff
@@ -123,12 +95,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
         End Function
 
         ' PERF: Cached values for GetDisplayAndInsertionText. Cuts down on the number of calls to ToMinimalDisplayString for large enums.
+        Private ReadOnly _gate As New Object()
```

Comment:
nit: a cleaner way to do this would be to just use a CWT map from INamedTypeSymbol to the cached info you care about for it.

Reply:
I thought about cwt but given the cache usage here it probably isn't worth it . So keeping it as is